File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/registry/entity/InterfaceMetaInfo.java
package org.javamaster.invocationlab.admin.service.registry.entity;

import com.google.common.collect.Sets;
import lombok.Data;

import java.util.Set;

/**
 * @author yudong
 * 接口及接口包含的方法及参数等元数据,从zk获取
 * 如果从其他地方拉取这个数据类型应该是兼容的
 */
@Data
public class InterfaceMetaInfo {
    String applicationName;
    String group;
    String version;
    String interfaceName;
    /**
     * 这个是zk拼接的完整地址:dubbo://192.....
     */
    String serviceAddr;
    Set<String> methodNames = Sets.newHashSet();
    Set<String> serverIps = Sets.newHashSet();
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/registry/entity/InterfaceMetaInfo.kt
package org.javamaster.invocationlab.admin.service.registry.entity

/**
 * @author yudong
 * 接口及接口包含的方法及参数等元数据,从zk获取
 * 如果从其他地方拉取这个数据类型应该是兼容的
 */

class InterfaceMetaInfo {
    var applicationName: String? = null
    var group: String? = null
    var version: String? = null
    var interfaceName: String? = null

    /**
     * 这个是zk拼接的完整地址:dubbo://192.....
     */
    var serviceAddr: String? = null
    var methodNames: Set<String> = mutableSetOf()
    var serverIps: Set<String> = mutableSetOf()
}
